rules/helm/def.bzl

```diff
@@ -63,6 +71,67 @@ def package(**kwargs):
         **kwargs
     )
 
+def _dependencies_impl(ctx):
+    # Get the attribute absolute paths.
+    helm = ctx.path(ctx.attr._helm)
+    chart_yaml = ctx.path(ctx.attr.chart_yaml)
+    requirements = ctx.path(ctx.attr.requirements)
+    requirements_lock = ctx.path(ctx.attr.requirements_lock)
+
+    # Symlink the required files into the cache.
+    ctx.symlink(chart_yaml, "Chart.yaml")
+    ctx.symlink(requirements, "requirements.yaml")
+    ctx.symlink(requirements_lock, "requirements.lock")
+
+    # Fetch the dependencies.
+    ctx.execute([helm, "dep", "up"])
+
+    # Create the workspace root BUILD.bazel.
+    ctx.file("BUILD.bazel", 'package(default_visibility = ["//visibility:public"])\n')
+
+    # Create the charts/BUILD.bazel exporting the fetched charts.
+    charts_build = """
+    package(default_visibility = ["//visibility:public"])
```

Comment:
If this ends up in the generated `BUILD.bazel`, I would prefer a template or something that is imported by this rule, at least is more clear and you carry the logic out in the code separately

Reply:
Hmm, so, I went to take a better look at the docs for the `repository_ctx`, and it turns out we can do what you asked: https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#read.

This file ends up being `@<workspace>//charts:BUILD.bazel`. So, if we use this rule like in the example I gave in the description:

```
helm_dependencies(
    name = "kubecf_helm_dependencies",
    chart_yaml = "//deploy/helm/kubecf:Chart.yaml",
    requirements = "//deploy/helm/kubecf:requirements.yaml",
    requirements_lock = "//deploy/helm/kubecf:requirements.lock",
)
```

Then, this file is `@kubecf_helm_dependencies//charts:BUILD.bazel`. It will export the `.tgz` charts downloaded with the `helm dep up` command.

Reply:
Fixed in https://github.com/cloudfoundry-incubator/kubecf/pull/707/commits/b733db8d722e8d910de3934ec3de54cd8ed107ca.